Weekly_Work/Week1/Code_changes-git/Metrics_Root_class.py

```diff
@@ -0,0 +1,202 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[1]:
+
+
+import json
+import datetime
+import matplotlib.pyplot as plt
+import pandas as pd
+
+
+# In[7]:
+
+
+class Metric():
+    
+    def __init__(self, path, since=None, until=None, source_code_exclude_list=None):
+
+        self.since = since
+        self.until = until
+        self.source_code_exclude_list = source_code_exclude_list
+        self.repo_urls = set()
+        
+        clean_commit_list = list()
+        clean_issue_list = list()
+        clean_pr_list = list()
+        
+        with open(path, 'r') as raw_data:
+            for line in raw_data:
+                line = json.loads(line)
+                self.repo_urls.add(line['origin'])
+                clean_line = dict()
+                if line['category'] == "commit":
+                    clean_line = self._clean_commit(line)
+
+                    if self._is_source_code(clean_line):
+                        clean_commit_list.append(clean_line)
+                    
+
+                elif line['category'] == "issue":
+                    if "pull_request" not in line['data']:
+                        clean_line = self._clean_issue(line)
+                        clean_issue_list.append(clean_line)
+                    else: continue
+                    
+
+                elif line['category'] == "pull_request":
+                    clean_line = self._clean_pr(line)
+                    clean_pr_list.append(clean_line)
+                        
+                        
+                self.clean_commit_df = pd.DataFrame(clean_commit_list)
```

Comment:
As said above, implementations for lists and dataframes would be in separate hierarchies of classes.

Reply:
The lists I used are only used for easily populating the dataframes. I did not try and mix two different implementations here. 
`self.clean_commit_df = pd.DataFrame(clean_commit_list)` is an example of what I said above.

The final implementations will be separate.

Reply:
OK with both being separate. I then understand that this is the implementation for the df class.

In any case, as I said above, I think the cleaning of the commit_list is something that shouldn't happen, at least at this level.